cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -287,7 +287,7 @@ public void delete(HttpRequest request, HttpResponder responder,
                      @PathParam("stream") String stream) throws Exception {
     Id.Stream streamId = Id.Stream.from(namespaceId, stream);
     checkStreamExists(streamId);
-
+    streamWriter.close(streamId);
```

Comment:
Correct, but we don't run distributed mode on Windows, do we? There is no easy fix for the distributed mode, we would have to through some coordination to make all stream writers close their files.